DetectorDescription/DDCMS/plugins/DDDefinitions2Objects.cc

```diff
@@ -445,30 +445,41 @@ template <> void Converter<DDLElementaryMaterial>::operator()(xml_h element) con
   if(  nullptr == mat ) {
     const char* matname = nam.c_str();
     double density      = xmat.density();
-    double atomicNumber = xmat.attr<double>(DD_CMU(atomicNumber));
+    int atomicNumber = xmat.attr<int>(DD_CMU(atomicNumber));
+    double atomicWeight = xmat.attr<double>(DD_CMU(atomicWeight));
     TGeoElementTable* tab = mgr.GetElementTable();
     TGeoMixture*      mix = new TGeoMixture(nam.c_str(), 1, density);
     TGeoElement*      elt = tab->FindElement(xmat.nameStr().c_str());
 
     printout(ns.context()->debug_materials ? ALWAYS : DEBUG, "DD4CMS",
-             "+++ Converting material %-48s  Density: %.3f.",
-             ('"'+nam+'"').c_str(), density);
+             "+++ Converting material %-48s  Atomic weight %.3f, Atomic number %u, Density: %.3f.",
+             ('"'+nam+'"').c_str(), atomicWeight, atomicNumber, density);
 
-    if( !elt )  {
-      printout(WARNING,"DD4CMS",
-               "+++ Converter<ElementaryMaterial> No element present with name:%s  [FAKE IT]",
-               matname);
-      int n = int(atomicNumber/2e0);
-      if( n < 2 ) n = 2;
-      elt = new TGeoElement(xmat.nameStr().c_str(),"CMS element",n,atomicNumber);
+    bool newMatDef = false;
+    
+    if(elt) {
+      // A is Mass of a mole in Geant4 units for atoms with atomic shell
+      printout(ns.context()->debug_materials ? ALWAYS : DEBUG, "DD4CMS",
+	       "    ROOT definition of %-50s Atomic weight %.3f, Atomic number %u, Number of nucleons %u",
+	       elt->GetName(), (elt->A())*g/mole, elt->Z(), elt->N() );
+      if(atomicNumber != elt->Z() || atomicWeight != (elt->A())*g/mole)
```

Comment:
It looks like there is more than one unit convention in this file. I'm not sure which convention `g` and `mole` follow. Is the ROOT convention the same as the CMS convention?
The `using namespace dd4hep;` and `using namespace geant_units::operators;` statements should probably be deleted or, at least, greatly limited in scope in order to require fully qualified names so it is more clear what convention each unit constant follows. `g` is particularly difficult -- how can I search the code to find where "g" is defined?
More comments about the different unit conventions in use would be helpful.

Reply:
@cvuosalo as far as I can see this should follow the DD4hep definition which looks a clone of CLHEP, @ianna may confirm. As this is CMS code in principle the code brought in DataFormats/Math could be used as well

Reply:
yes, I'll stick with the DD4hep one here. There were multiple discussions among DD4hep developers. I hope, we'll converge on something common eventually.

Reply:
@fabiocos: I was not suggesting changing which unit conventions are used in this file. I was just requesting they be clearly commented. 
This file contains these two lines:
```
using namespace dd4hep;
using namespace geant_units::operators;
```
From what I found in `DD4hep/Evaluator/DD4hepUnits.h`, the DD4hep units convention does not match the Geant units convention. Because of this incompatibility, it is risky to use both namespaces at once because future developers and reviewers may get confused as to which units convention a variable or constant follows.
I would delete those two `using namespace` statements and then write the conversion as:
```
double dd4hepAtomMass = elt->A() * dd4hep::g / dd4hep:mole;
// Converting elt->A() in grams/mole in the Geant units convention to the
// DD4hep units connvention.
```
I would also add a comment at the top of the file saying:
```
/**** CAUTION: This file uses both the CMS and Geant units conventions.   ****/
```
If these changes can't be made now, I will make them myself in a later PR.